### PR TITLE
Worker Docker release build without using cache

### PR DESCRIPTION
**Pull request by Traf333**
this PR is a follow up for #2313. At regular process using cache for worker build helps us to speed up daily workflow, but in case of preparing releases which happen not so often, we decided to avoid using cache for building docker images

### CODE REVIEW

File: .github/workflows/create-release-draft.yml [hunk 1]
```diff
@@ -173,13 +173,9 @@ jobs:
...
-        with:
-          ref: ${{ env.RELEASE_TAG }}
-          fetch-depth: 0
-      - name: Set env
-        run: |
-          WORKER_DOCKER_TAG=$(echo ${{ env.RELEASE_TAG }} | sed 's/.*\(w.*\)/\1/;s/w/v/')
-          echo "WORKER_DOCKER_TAG=$WORKER_DOCKER_TAG" >> $GITHUB_ENV
+        # with:
+        # ref: ${{ env.RELEASE_TAG }}
+        # fetch-depth: 0
```
Comment (Traf333) [inline .github/workflows/create-release-draft.yml]: I'll return it back once build release works fine

Comment (Traf333) [reply .github/workflows/create-release-draft.yml]: https://github.com/litentry/litentry-parachain/actions/runs/7669315590
looks like it works fine

File: tee-worker/build.Dockerfile [hunk 2]
```diff
@@ -47,17 +47,25 @@ ENV WORKER_MODE=$WORKER_MODE_ARG
...
+
 ARG FINGERPRINT=none
 
 WORKDIR $HOME/tee-worker
 COPY . $HOME
 
 RUN \
+  if [ "$IMAGE_FOR_RELEASE" = "true" ]; then \
+  echo "Omit cache for release image"; \
+  make; \
```
Comment (Kailai-Wang) [inline tee-worker/build.Dockerfile]: Ideally with proper indention :)

File: .github/workflows/create-release-draft.yml (hunk 1 above)
Comment (Kailai-Wang) [reply .github/workflows/create-release-draft.yml]: Hmm shall we revert the comment?

File: .github/workflows/create-release-draft.yml [hunk 3]
```diff
@@ -207,6 +208,7 @@ jobs:
           build-args: |
             WORKER_MODE_ARG=sidechain
             ADDITIONAL_FEATURES_ARG=
+            IMAGE_FOR_RELEASE=true
```
Comment (Kailai-Wang) [inline .github/workflows/create-release-draft.yml:211]: Acutally we don't even need to build `local-builder:latest` here

**Review by Kailai-Wang — COMMENTED** [review]
Thank you!

I personally prefer a separate Dockerfile where we don't need everything in local-builder (e.g. sccache) , but I'm fine with this version too :D

File: .github/workflows/create-release-draft.yml (hunk 3 above)
Comment (Traf333) [reply .github/workflows/create-release-draft.yml:211]: I was thinking that we need it for the next step when we build `worker` and `cli` separately

**Comment by Traf333** [comment]
> I personally prefer a separate Dockerfile where we don't need everything in local-builder (e.g. sccache) , but I'm fine with this version too

 @Kailai-Wang , that's valid point and I was thinking of supporting two version of docker file over one with just minor "if" statement, but than I realised that later we may introduce new changes like including `extract_identity` binary or `ts-tests` folder to register worker to the docker image, which would require further modification. This thoughts lean me towards having single file for now

File: .github/workflows/create-release-draft.yml (hunk 3 above)
Comment (Kailai-Wang) [reply .github/workflows/create-release-draft.yml:211]: Oh yes I was thinking of using another Dockerfile, sorry

File: .github/workflows/create-release-draft.yml [hunk 4]
```diff
@@ -174,12 +174,8 @@ jobs:
...
         uses: actions/checkout@v4
         with:
-          ref: ${{ env.RELEASE_TAG }}
-          fetch-depth: 0
-      - name: Set env
-        run: |
-          WORKER_DOCKER_TAG=$(echo ${{ env.RELEASE_TAG }} | sed 's/.*\(w.*\)/\1/;s/w/v/')
-          echo "WORKER_DOCKER_TAG=$WORKER_DOCKER_TAG" >> $GITHUB_ENV
+        ref: ${{ env.RELEASE_TAG }}
+        fetch-depth: 0
```
Comment (Kailai-Wang) [inline .github/workflows/create-release-draft.yml]: The indention still doesn't seem right

**Review by Kailai-Wang — APPROVED** [review]
Looks good - a small indention problem

**Pull request merged**